### PR TITLE
Adds Heroku

### PR DESCRIPTION
**Pull request by deadlybutter**
This pull request allows phoenix-next to run on Heroku, additionally it adds an `app.json` file so review apps can be generated on the fly for pull requests.

- Adds Procfile -- Used by Heroku to start the process
- Adds app.json -- This has a few env vars set, but most are copied from the staging dyno at build time. Additionally it defines a mysql service and a post deploy script
- Adds setup.sh -- Runs migrations, seed & builds fender assets
- Changed database config -- This was needed to support mysql strings that are provided from the cloud service

cc @blisteringherb

### CODE REVIEW

**Comment by deadlybutter** [comment]
(Im also going to squash these commits, ha....)

File: app.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "phoenix-next",
+  "description": ":sparkles:",
+  "scripts": {
+    "postdeploy": "sh bin/setup.sh"
```
Comment (DFurnes) [inline app.json]: Can we move this into `bootstrap/` so that we don't add a new top-level directory?

File: bin/setup.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+php artisan migrate --seed --force
+npm install --only=dev
```
Comment (DFurnes) [inline bin/setup.sh]: Do we want this `--only=dev` flag?

File: config/database.php [hunk 3]
```diff
@@ -1,5 +1,19 @@
 <?php
 
+$mysql_host = env('DB_HOST', 'localhost');
+$mysql_port = env('DB_PORT', '3306');
+$mysql_database = env('DB_DATABASE', 'forge');
+$mysql_username = env('DB_USERNAME', 'forge');
+$mysql_password = env('DB_PASSWORD', '');
+
+if (env('CLEARDB_DATABASE_URL')) {
```
Comment (DFurnes) [inline config/database.php]: Can you add a comment above here explaining that this is used for Heroku environments?

File: app.json (hunk 1 above)
Comment (deadlybutter) [inline app.json]: fine with me!

File: bin/setup.sh (hunk 2 above)
Comment (deadlybutter) [inline bin/setup.sh]: the heroku npm buildback does the normal npm install, but it leaves out the dev dependencies (so yes, intentional)

File: config/database.php (hunk 3 above)
Comment (deadlybutter) [inline config/database.php]: 👍

File: bin/setup.sh (hunk 2 above)
Comment (DFurnes) [inline bin/setup.sh]: Ah, that makes sense! Can you add a `# comment` above explaining that?

Comment (deadlybutter) [inline bin/setup.sh]: sure thing!

**Comment by deadlybutter** [comment]
Seems like everything still works after those changes, let me know if the comments help clear things up

**Comment by DFurnes** [comment]
Woohoo! 🎉 This is so cool:

screen shot 2017-02-07 at 5 22 08 pm

**Pull request merged**